lib/xdr/accounts/thresholds.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Stellar.XDR.Thresholds do
+  @moduledoc """
+  Representation of an AccountEntry's Thresholds.
+  Thresholds stores unsigned bytes: [weight of master|low|med|high]
+  """
+  alias Stellar.XDR.Opaque4
+
+  @behaviour XDR.Declaration
+
+  @type thresholds :: [
+          master_weight: non_neg_integer(),
+          low: non_neg_integer(),
+          med: non_neg_integer(),
+          high: non_neg_integer()
+        ]
+
+  @type t :: %__MODULE__{thresholds: thresholds}
+
+  defstruct [:thresholds]
+
+  @spec new(master_weight: byte(), low: byte(), med: byte(), high: byte()) :: t()
+  def new([master_weight: _master_weight, low: _low, med: _med, high: _high] = thresholds),
+    do: %__MODULE__{thresholds: thresholds}
+
+  @impl true
+  def encode_xdr(%__MODULE__{
```

Comment:
@JuanCamiloPerilla this pattern matching will be not required if we just validate the thresholds list when calling the `new` function

Reply:
The module's struct was flattened, so the variables in the pattern match are now required within the function.